include/pf-applications/lac/dynamic_block_vector.h

```diff
@@ -58,6 +58,9 @@ namespace dealii
                 blocks[b] = std::make_shared<BlockType>();
               block(b).reinit(V.block(b), true);
               block(b) = V.block(b);
+              
+              if (V.block(b).has_ghost_elements())
+                block(b).update_ghost_values();
```

Comment:
It probably also makes sense to keep ghosts in copies if they initially were there. But this change is arguable.